lib/airbrake-ruby/sync_sender.rb

```diff
@@ -33,18 +21,15 @@ def initialize(config)
     # @param [Airbrake::Notice] endpoint
     # @return [Hash{String=>String}] the parsed HTTP response
     def send(notice, endpoint = @config.endpoint)
-      response = nil
       req = build_post_request(endpoint, notice)
       https = build_https(endpoint)
 
-      begin
-        response = https.request(req)
-      rescue *HTTP_ERRORS => ex
-        @config.logger.error("#{LOG_LABEL} HTTP error: #{ex}")
-        return
-      end
-
+      response = https.request(req)
       Response.parse(response, @config.logger)
+
+    rescue => ex
```

Comment:
I think its true that almost all of the exceptions will happen in https.request, but if any other RuntimeErrors occur, then a problem cascades upwards for users of the async sender. I believe it works like this:

1. If any exception here is uncaught
2. it causes the async worker thread(s) to crash.
3. and that causes AsyncSender#has_workers? to return false

Which means we fall back to the default sync sender which we're trying to avoid.

So we can rescue only https.request, but then I think we'd need another way to make sure the workers can't crash or some configuration to ensure async behavior such as #50. The intent I think is to have some way to make sure that an app (if the authors like) is only sending via background threads.